src/js/cf-expandables.js

```diff
@@ -5,110 +5,168 @@
  * A public domain work of the Consumer Financial Protection Bureau
  */
 
-(function( $ ) {
+( function( $ ) {
+
+  'use strict';
+
+  var _extend = $.extend;
+
+  var Expandable = {
+
+    defaultProperties: {
+      isInAccordion: false,
+      isExpanded:    false,
+      expandedClass: 'expandable__expanded'
+    },
+
+    // Elements that will be cached.
+    ui: {
+      cueOpen:  '.expandable_cue-open',
+      cueClose: '.expandable_cue-close',
+      content:  '.expandable_content',
+      target:   '.expandable_target'
+    },
+
+    create: function create( options ) {
+      var properties = _extend( true, {}, Expandable.defaultProperties, options );
+
+      // Using $.extend to create an instance of the Expandable.
+      // Extending dom elements is generally bad and should be
+      // avoided. `this` is referring to the dom element.
+      _extend( this, Expandable, properties );
+      this.init( );
+    },
+
+    init: function init() {
+      this.isInAccordion = Boolean( this.$el.parents( '.expandable-group' )
+                           .data( 'accordion' ) );
+      this.initUI();
+      this.initEvents();
+    },
+
+    destroy: function destroy() {
+      if ( this.$el ) {
+        this.$el.remove();
+      }
+    },
 
-  $.fn.expandable = function( userSettings ) {
+    initUI: function initUI() {
+      var $el = this.$el;
 
-    return $( this ).each(function() {
+      for ( var uiKey in this.ui ) {
+        if ( this.ui.hasOwnProperty( uiKey ) ){
+          this['$' + uiKey] = $el.find( this.ui[uiKey] + ':first' );
+        }
+      }
 
-      var $this = $( this ),
-          $target = $this.find('.expandable_target').not( $this.find('.expandable .expandable_target') ),
-          $cueOpen = $this.find('.expandable_cue-open').not( $this.find('.expandable .expandable_cue-open') ),
-          $cueClose = $this.find('.expandable_cue-close').not( $this.find('.expandable .expandable_cue-close') ),
-          $content = $this.find('.expandable_content').not( $this.find('.expandable .expandable_content') ),
-          $groupParent = $this.parents('.expandable-group'),
-          accordion = $groupParent.length > 0 && $groupParent.data('accordion');
+      this.$target.attr( 'aria-controls', this.$content.attr( 'id' ) );
 
-      if ( accordion ) {
-        var $siblings = $this.siblings('.expandable');
+      if ( $el.hasClass( this.expandedClass ) ) {
```

Comment:
Is this storing state or just checking to see if an element is an expandable? If it's used for state, could we store that outside of the DOM?